packages/protocol/test/StakingRewards.test.ts

```diff
@@ -241,82 +282,199 @@ describe("StakingRewards", function () {
+          from: anotherUser,
+        })
+        const startTime = await time.latest()
+
+        const tokenId = await stake({amount: curveLPAmount, positionType: StakedPositionType.CurveLP, from: investor})
+        const timeDiff = (await time.latest()).sub(startTime)
+
+        await advanceTime({seconds: yearInSeconds})
+
+        // investor owns 1/5 of the staked supply and therefore should receive 1/5
+        // of the disbursed rewards
+        await stakingRewards.getReward(tokenId, {from: investor})
+        let expectedRewards = maxRate.mul(yearInSeconds.sub(timeDiff)).div(new BN(5))
+        expect(await gfi.balanceOf(investor)).to.bignumber.equal(expectedRewards)
+
+        // anotherUser owns 4/5 of the staked supply and therefore should receive 4/5
+        // of the disbursed rewards
+        await stakingRewards.getReward(anotherUserToken, {from: anotherUser})
+        const rewardsWhenOnlyAnotherUserWasStaked = maxRate.mul(timeDiff)
+        const rewardsWhenInvestorWasStaked = maxRate.mul(yearInSeconds.sub(timeDiff)).mul(new BN(4)).div(new BN(5))
+        expectedRewards = rewardsWhenOnlyAnotherUserWasStaked.add(rewardsWhenInvestorWasStaked)
+        expect(await gfi.balanceOf(anotherUser)).to.bignumber.equal(expectedRewards)
+      })
+
+      it("splits rewards amongst stakers proportional to their stakes with different exchange rates", async () => {
+        stakingRewards._setBaseTokenExchangeRate(StakedPositionType.CurveLP, new BN(2).mul(new BN(String(1e18))))
+
+        // anotherUser stakes 2x more FIDU tokens than investor in Curve LP tokens
+        const anotherUserToken = await stake({
+          amount: curveLPAmount.mul(new BN(2)),
+          from: anotherUser,
+        })
+        const startTime = await time.latest()
+
+        const tokenId = await stake({amount: curveLPAmount, positionType: StakedPositionType.CurveLP, from: investor})
+        const timeDiff = (await time.latest()).sub(startTime)
+
+        await advanceTime({seconds: yearInSeconds})
+
+        // investor owns 1/2 of the staked supply and therefore should receive 1/2
+        // of the disbursed rewards
+        await stakingRewards.getReward(tokenId, {from: investor})
+        let expectedRewards = maxRate.mul(yearInSeconds.sub(timeDiff)).div(new BN(2))
+        expect(await gfi.balanceOf(investor)).to.bignumber.equal(expectedRewards)
+
+        // anotherUser owns 1/2 of the staked supply and therefore should receive 1/2
+        // of the disbursed rewards
+        await stakingRewards.getReward(anotherUserToken, {from: anotherUser})
+        const rewardsWhenOnlyAnotherUserWasStaked = maxRate.mul(timeDiff)
+        const rewardsWhenInvestorWasStaked = maxRate.mul(yearInSeconds.sub(timeDiff)).div(new BN(2))
+        expectedRewards = rewardsWhenOnlyAnotherUserWasStaked.add(rewardsWhenInvestorWasStaked)
+        expect(await gfi.balanceOf(anotherUser)).to.bignumber.equal(expectedRewards)
+      })
     })
```

Comment:
I see that this test mixes fidu and curve LP stakers, which is great. Can we add the following test too:

* A test showing that the curve LP test uses the effective multiplier